src/assets/toolkit/scripts/lib/component/float-label.js

```diff
@@ -2,26 +2,32 @@
 
 export class FloatLabel {
   /**
+   * @see https://mathiasbynens.be/notes/oninput
    * @param {DOM} element
    * @param {Object} options
    * @param {String} options.inputElement - Reference to input within element.
-   * @param {String} options.eventName - Event that will trigger updates.
    * @param {String} options.className - Class applied to `element` when `inputElement` is empty.
    */
   constructor(element, {
     inputElement = element.querySelector('input, textarea'),
-    eventName = 'input',
     className = 'is-empty'
   } = {}) {
     // Assign properties
     Object.assign(this, {
       element,
       inputElement,
-      eventName,
       className
     });
-    // Listen for event and update
-    this.element.addEventListener(this.eventName, this.update.bind(this));
+    // Fallback event handler
+    var keyUpHandler = () => this.update();
+    // Ideal event handler, will remove the other one if supported
+    var inputHandler = () => {
+      this.element.removeEventListener('keyup', keyUpHandler);
```

Comment:
> But that seemed like extra complexity, and Mathias's stuff (who was the source for this) tends to be pretty performant.

Very interesting. I like the spirit of that, but you are right, seems more complex than it needs to be. 😁 

What you have now is _totally_ fine. Yeah, curious if @erikjung has any new insights. 😄